File: 1013-partition-array-into-three-parts-with-equal-sum/1013-partition-array-into-three-parts-with-equal-sum.py
class Solution(object):
    def canThreePartsEqualSum(self, arr):
        eachsum = sum(arr)/3
        s = 0
        idx = 0 
        count = 0
        while idx < len(arr):
            s += arr[idx]
            idx += 1
            if s == eachsum and count<3:
                s = 0
                count += 1
        if count == 3:
            return True
        else:
            return False

File: 1013-partition-array-into-three-parts-with-equal-sum/1013-partition-array-into-three-parts-with-equal-sum.c
bool canThreePartsEqualSum(int* arr, int arrSize) {
    int totalSum = 0;
    for (int i=0; i<arrSize; i++) {
        totalSum += arr[i];
    }
    if (totalSum % 3 != 0) return false;
    int count = 0;
    int part = 0;
    for (int i=0; i<arrSize; i++) {
        part += arr[i];
        if (part == totalSum/3) {
            part = 0;
            count += 1;
        }
    }

    if (count==3) return true;
    else return false;
}